### ISSUE TITLE
ResolvePackageDependencies does not handle native winmds correctly

### ISSUE THREAD

**Issue by ericstj** [dotnet/sdk#568]
See https://github.com/NuGet/NuGet.BuildTasks/blob/4a57a0ef4bee31f122d50a39fe3dddd4ca03fb07/src/Microsoft.NuGet.Build.Tasks/ResolveNuGetPackageAssets.cs#L380 for an example of what needs to be done.

/cc @jasonmalinowski @natidea @nguerrera

**Comment by ericstj** [comment]
For that matter you should probably examine all the differences between ResolveNuGetPackageAssets and ResolvePackageDependencies to make sure we aren't missing any functionality...

**Comment by natidea** [comment]
Thanks for the note @ericstj . When we designed `ResolvePackageDependencies` task, we intentionally limited its scope to simply raising the assets file to MSBuild items that can be further manipulated downstream. In other words, the goal is to replace `ResolveNuGetPackageAssets` with multiple smaller tasks that compose better. (We have also initially limited our scope to functionality needed for the next .NET Core tooling release.)

As an example, `ResolveNuGetPackageAssets` has some functionality for [producing content assets](https://github.com/NuGet/NuGet.BuildTasks/blob/4a57a0ef4bee31f122d50a39fe3dddd4ca03fb07/src/Microsoft.NuGet.Build.Tasks/ResolveNuGetPackageAssets.cs#L533) and [handling preprocessor values](https://github.com/NuGet/NuGet.BuildTasks/blob/4a57a0ef4bee31f122d50a39fe3dddd4ca03fb07/src/Microsoft.NuGet.Build.Tasks/Preprocessor.cs#L15). In the SDK, we moved this functionality to a downstream [task](https://github.com/dotnet/sdk/blob/a700d03b9e671f70ae81de84803f3e2e2cdf74b8/src/Tasks/Microsoft.NET.Build.Tasks/ProduceContentAssets.cs#L19) and a separate [MSBuild target](https://github.com/dotnet/sdk/blob/a700d03b9e671f70ae81de84803f3e2e2cdf74b8/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets#L198).

What is the timeframe in which support for native winmds is needed? I think the correct solution will involve new tasks/targets, and probably a new targets file.

/cc @srivatsn @davkean
